lib/rtti_utils.h

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2024-present Intel Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/// @file
+/// @brief Utilities that help with use of custom-RTTI-enabled classes.
+
+#ifndef LIB_RTTI_UTILS_H_
+#define LIB_RTTI_UTILS_H_
+
+#include <type_traits>
+
+#include "rtti.h"
+
+namespace RTTI {
+
+/// A trait that check T is custom-RTTI-enabled. Works just like standard property type traits.
+/// One would normally use the _v variant.
+template <typename T>
+struct has_rtti : std::is_base_of<RTTI::Base, T> {};
```

Comment:
There are two pieces of RTTI in general:
 - One needs to be derived from `RTTI:Base` (plus all its bases must be derived from `RTTI::Base`)
 - One has appropriate methods implementation (e.g. via `DECLARE_TYPEINFO`-kind macro)

I think it's ok to check just for the base class here – `TypeInfo` will check for the whole set of constraints anyway should the RTTI methods are invoked.

Reply:
Yes, I'd say being derived from `RTTI::Base` without the proper macros is definitely a bug, so firing a static_assert instead of the proper SFINAE-enabling failure in that case seems fine to me.